SU2_CFD/src/fluid/CFluidFlamelet.cpp

```diff
@@ -0,0 +1,150 @@
+    cout << "*****************************************" << endl;
+    cout << "***   initializing the lookup table   ***" << endl;
+    cout << "*****************************************" << endl;
+  }
+
+  table_scalar_names.resize(n_scalars);
+  table_scalar_names[I_ENTH] = "EnthalpyTot";
+  table_scalar_names[I_PROGVAR] = "ProgressVariable";
+  /*--- auxiliary species transport equations---*/
+  for (size_t i_aux = 0; i_aux < n_user_scalars; i_aux++) {
+    table_scalar_names[n_control_vars + i_aux] = config->GetUserScalarName(i_aux);
+  }
+
+  look_up_table = new CLookUpTable(config->GetFileNameLUT(), table_scalar_names[I_PROGVAR], table_scalar_names[I_ENTH]);
+
+  n_lookups = config->GetNLookups();
+  table_lookup_names.resize(n_lookups);
+  for (int i_lookup = 0; i_lookup < n_lookups; ++i_lookup) {
+    table_lookup_names[i_lookup] = config->GetLUTLookupName(i_lookup);
+  }
+
+  Pressure = value_pressure_operating;
+
+  PreprocessLookUp();
+}
+
+CFluidFlamelet::~CFluidFlamelet() {
+  delete look_up_table;
+}
+
+void CFluidFlamelet::SetTDState_T(su2double val_temperature, const su2double* val_scalars) {
+  su2double val_enth = val_scalars[I_ENTH];
+  su2double val_prog = val_scalars[I_PROGVAR];
+
+  /*--- Add all quantities and their names to the look up vectors. ---*/
+  look_up_table->LookUp_XY(varnames_TD, val_vars_TD, val_prog, val_enth);
+
+  Temperature = val_vars_TD[GetIdx(varnames_TD,"Temperature")];
+  Density = val_vars_TD[GetIdx(varnames_TD,"Density")];
+  Cp = val_vars_TD[GetIdx(varnames_TD,"Cp")];
+  Mu = val_vars_TD[GetIdx(varnames_TD,"ViscosityDyn")];
+  Kt = val_vars_TD[GetIdx(varnames_TD,"Conductivity")];
+  mass_diffusivity = val_vars_TD[GetIdx(varnames_TD,"DiffusionCoefficient")];
+  molar_weight = val_vars_TD[GetIdx(varnames_TD,"MolarWeightMix")];
+
+  /*--- Compute Cv from Cp and molar weight of the mixture (ideal gas). ---*/
+  Cv = Cp - UNIVERSAL_GAS_CONSTANT / molar_weight;
+}
+
+/* --- Total enthalpy is the transported variable, but we usually have temperature as a boundary condition,
+       so we do a reverse lookup */
+unsigned long CFluidFlamelet::GetEnthFromTemp(su2double& val_enth, const su2double val_prog, const su2double val_temp,
+                                              const su2double initial_value) {
+
+  /*--- convergence criterion for temperature in [K], high accuracy needed for restarts. ---*/
+  su2double delta_temp_final = 0.001;
+  su2double enth_iter = initial_value;
+  su2double delta_enth;
+  su2double delta_temp_iter = 1e10;
+  unsigned long exit_code = 0;
+  const int counter_limit = 1000;
+
+  int counter = 0;
+  while ((abs(delta_temp_iter) > delta_temp_final) && (counter++ < counter_limit)) {
+    /*--- Add all quantities and their names to the look up vectors. ---*/
+    look_up_table->LookUp_XY(varnames_TD, val_vars_TD, val_prog, enth_iter);
+    Temperature = val_vars_TD[GetIdx(varnames_TD,"Temperature")];
+    Cp = val_vars_TD[GetIdx(varnames_TD,"Cp")];
```

Comment:
This is very inefficient. If the variables are always in the same position (which they are judging from PreprocessLookUp) use an enum Temperature = 0, etc. instead of this search into varnames_TD every time.

Reply:
OK, I introduced enums.